prometheus/vec.go

```diff
@@ -20,12 +20,19 @@ import (
 	"github.com/prometheus/common/model"
 )
 
-// metricVec is a Collector to bundle metrics of the same name that differ in
-// their label values. metricVec is not used directly (and therefore
-// unexported). It is used as a building block for implementations of vectors of
-// a given metric type, like GaugeVec, CounterVec, SummaryVec, and HistogramVec.
-// It also handles label currying.
-type metricVec struct {
+// MetricVec is a Collector to bundle metrics of the same name that differ in
+// their label values. MetricVec is not used directly but as a building block
+// for implementations of vectors of a given metric type, like GaugeVec,
+// CounterVec, SummaryVec, and HistogramVec. Is is exported so that it can be
+// used for custom Metric implementations. To create a FooVec for custom Metric
```

Comment:
Super useful comment. I would suggesting adding new line to separate this tip for `Vec` implementation for custom type with upper summary?

Reply:
Sure. Done.